core/corehttp/gateway_handler.go

```diff
@@ -133,6 +133,7 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	}
 
 	w.Header().Set("X-IPFS-Path", urlPath)
+	w.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
strongly disagree.

---

the IPFS gateway is an HTTP server like any other. People will run into annoying configuration problems with it.

ideally for us, IPFS itself would remain completely unaware of any garbage like having to worry about CORS headers, and instead provide the user the tooling necessary to solve their own problem, whatever it might be. different deployments will call for different things.

> adding configuration options makes the program more complex both to use and maintain.

so does adding some options and not others. it creates management complexity. forwarding configurations is a unix-tool thing to do, too.

> [softcoding](
> http://thedailywtf.com/articles/Soft_Coding).

i disagree with much in this article. a counter example is auth tokens and private keys.

it's also not very applicable, as this is not "specific, soft coded business logic complicating one group's deployments", but rather facilities in a _general_ tool dealing with the concerns of _many different types_ of users with _many different types_ of concerns. In the quest of not solving everybody's problems, people tend to solve a _few_ people's problems and leave it at that. Instead, we can sidestep the issue and _let the users_ solve their own problems.

> I think that adding arbitrary headers should be handled by another tool (like nginx),

Sure, and this applies to CORS headers too.

But it's a common enough thing that it was worth considering